### PR TITLE
Switch to Gatsby + Vercel for better theme/features

### PR DESCRIPTION
**Pull request by craigmdennis**
# Motivation
Turning our existing content into more of a 'behind-the-scenes of building Close' type site which could encompass Engineering posts, Design posts, how we make some product decisions, etc. Beyond blog posts, it could also become a helpful tool for recruiting into the various roles across eng/PM/Design. 

## To Do
- [x]  Resolve merge conflicts
- [x] Add Close logo
- [x] Add a link to close.com (in Footer)
- [x] Add post author names
- [x] Branch previews with Vercel (see the bottom of this PR)
- [x] Reorganize posts back into folders with images
- [x] Change references to site domain to `making.close.com`
- [x] Add Segment tracking script (via Gatsby plugin)

## Once Merged
- [ ] Make sure Vercel has built and deployed to 'production'
- [ ] Update domain in Vercel to engineering.close.com
- [ ] [Update DNS to point to Vercel](https://vercel.com/docs/custom-domains)

## What this PR does
- Uses Gatsby instead of Jekyll
- Uses a more design-friendly theme
- Builds & deploys using Vercel (including branch/PR deploy previews)
- Is ready for integrating with a headless CMS
- FIles are still markdown and still use frontmatter as previous

## Things to consider:
- Pagination is not included yet
- Tags have been added to the frontmatter but pages aren't yet generated, tags are just displayed on posts.

CleanShot 2020-09-25 at 15 47 26
CleanShot 2020-09-25 at 15 47 37

## What's Next?
- Tag pages
- Author pages
- Pagination
- Headless or GIT based CMS

---

The reason for _not_ using Netlify CMS is the way they preview content and use images. The images are uploaded and added to the markdown frontmatter with absolute paths. Gatsby image requires relative paths. There are some plugins but none seemed to work correctly.

Alternatives are:
- https://www.netlifycms.org/docs/netlify-large-media/ (would be useful if we use Netlify for hosting)
- https://www.netlifycms.org/docs/cloudinary/
- https://www.netlifycms.org/docs/uploadcare/

### CODE REVIEW

File: _data/posts/2013-05-02-sweating-the-ui-ux-details-in-close-emails.md [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 layout: post
-title: "Sweating the UI/UX details in Close: Emails & email addresses"
+title: 'Sweating the UI/UX details in Close: Emails & email addresses'
 date: 2013-05-02
-permalink: posts/sweating-the-ui-ux-details-in-close-emails
+permalink: /posts/sweating-the-ui-ux-details-in-close-emails
 author: Phil Freo
+thumbnail: ''
+metaDescription: ''
+tags: [engineering]
```
Comment (philfreo) [inline _data/posts/2013-05-02-sweating-the-ui-ux-details-in-close-emails.md:9]: Not sure exactly what all tags we want, but this one is more `product` than `engineering` focused.

File: src/pages/404.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
```
Comment (philfreo) [inline src/pages/404.js:1]: delete `src/images/icon.png` (file above this one)

**Review by philfreo — COMMENTED** [review]
Great stuff, thanks for moving this forward so far! Quick comments before I'm OOO:

- Your token link is broken, but can we use a Deploy Key instead of a user-specific one? (https://docs.github.com/en/developers/overview/managing-deploy-keys#deploy-keys)
  - Alternatively, if we want Vercel anyway for branch previews, then we might as well deploy this fully to Vercel (instead of GitHub Pages), and in that case then I don't think you need any GitHub token at all. Bart can help you get that setup since he already has a token for Vercel+GH. I've made both of you admins of this repo now too.
- Are we sure we don't want to keep post-specific folders for images (even if they all get moved inside `/static/assets/uploads` to keep track of which post-specific images go where?)

File: _data/posts/2013-05-02-sweating-the-ui-ux-details-in-close-emails.md (hunk 1 above)
Comment (craigmdennis) [reply _data/posts/2013-05-02-sweating-the-ui-ux-details-in-close-emails.md:9]: Good catch. Tags will have their own pages. And we might call out specific tags some global navigation somewhere. I'll change this one now.

**Comment by craigmdennis** [comment]
> Are we sure we don't want to keep post-specific folders for images (even if they all get moved inside /static/assets/uploads to keep track of which post-specific images go where?)

This is an artefact of using a 'media library' while experimenting with Netlify CMS. I'll revert back to folders in the mean time for easier linking.

File: README.md [hunk 3]
```diff
@@ -1,31 +1,44 @@
...
-docker-compose up -d
 ```
 
-Open:
-
-https://localhost:4000/
+\*while not necessary for creating the post (which relies on the slug in the frontmatter) it helps file organization.
 
-When done, shut it down with `docker-compose down`.
+## Developing the site
```
Comment (philfreo) [inline README.md:33]: We should still include a Docker-based set of instructions for people that don't want to mess with node versions etc on their local machine. However that can come later after this is merged.

File: _includes/footer.html [hunk 4]
```diff
@@ -1,28 +0,0 @@
...
-    </div>
-  
-  </footer>
-
-  <script>
-    !function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t,e){var n=document.createElement("script");n.type="text/javascript";n.async=!0;n.src="https://cdn.segment.com/analytics.js/v1/"+t+"/analytics.min.js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(n,a);analytics._loadOptions=e};analytics.SNIPPET_VERSION="4.1.0";
-    analytics.load("Ak24tvTW8u6qHxyE4Gk5mOXXXdmsxQHt");
-    analytics.page();
-    }}();
-  </script>
```
Comment (philfreo) [inline _includes/footer.html:27]: We should keep this analytics tracking script on the new site. Can you add that?

File: _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md [hunk 5]
```diff
@@ -114,86 +116,95 @@ Then...
 Ours ended up looking something like this:
 
 ```js
-const REDIRECTS_DATA_REGION = 'us-east-1';
-const REDIRECTS_DATA_BUCKET = 'example.com';
-const REDIRECTS_DATA_OBJECT = 'config/redirects.json';
+const REDIRECTS_DATA_REGION = 'us-east-1'
```
Comment (philfreo) [inline _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md:119]: Here and elsewhere: We shouldn't be stripping semicolons from our JavaScript examples

File: _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md [hunk 6]
```diff
@@ -42,7 +44,7 @@ Here's an example of that XML format (configured in AWS Console under your S3 bu
     <Redirect>
       <Protocol>https</Protocol>
       <HostName>close.com</HostName>
-      <ReplaceKeyWith/>
+      <ReplaceKeyWith />
```
Comment (philfreo) [inline _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md:47]: I'm not sure if this new change is correct or not but I'd say don't change it unless you're sure it is.

File: src/components/navigation.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Link } from 'gatsby';
+
+const Navigation = () => (
```
Comment (philfreo) [inline src/components/navigation.js:4]: Remove this if it's not used yet?

**Comment by philfreo** [comment]
What's the branch preview password? I tried `branchpreviewsareverynice` (like we use on closeio) but that didn't work.

File: src/pages/404.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+          <path
+            fill-rule="evenodd"
+            clip-rule="evenodd"
+            d="M2.74884 14.6663C3.73056 16.6421 5.76939 18 8.12537 18H16.1254C18.5654 18 20.6652 16.5435 21.6029 14.4525C22.3722 13.9093 22.8746 13.0133 22.8746 12C22.8746 10.9867 22.3722 10.0907 21.6029 9.54753C20.6652 7.45651 18.5654 6 16.1254 6H8.12537C5.76939 6 3.73056 7.3579 2.74884 9.33375C1.78448 9.83263 1.12537 10.8393 1.12537 12C1.12537 13.1607 1.78448 14.1674 2.74884 14.6663ZM8.12537 8H16.1254C17.5088 8 18.7282 8.70234 19.4465 9.76991C19.7227 10.4593 19.8746 11.2119 19.8746 12C19.8746 12.7881 19.7227 13.5407 19.4465 14.2301C18.7282 15.2977 17.5088 16 16.1254 16H8.12537C5.91623 16 4.12537 14.2091 4.12537 12C4.12537 9.79086 5.91623 8 8.12537 8Z"
+            fill="currentColor"
+          />
+        </svg>
+        <p>
+          Either you are in a wrong page or you have lost. Lets go back{' '}
+          <Link to="/">home</Link> safely
```
Comment (philfreo) [inline src/pages/404.js:42]:
```suggestion
          This page does not exist. Head to our <Link to="/">home page</Link> or let us know if something seems wrong.
```

**Comment by craigmdennis** [comment]
It's in the 1Password vault: `deathly-lime-fennel-aura`

File: .eslintrc.json [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+    "analytics": true,
+    "Rollbar": true,
+    "mixpanel": true,
+    "Twilio": true,
+    "jQuery": true,
+    "S3Upload": true,
+    "Mousetrap": true,
+    "StripeCheckout": true,
+    "google": true,
+    "dagreD3": true,
```
Comment (philfreo) [inline .eslintrc.json:43]: It's good that you tried to follow the same as our closeio repo, but some of this definitely doesn't belong (and will ultimately be cleaned up from the UI repo too). @lukasmlady Can you suggest a clean eslintrc that matches what we'd want for a new repo?

(Also related to my semicolon comment below)

**Review by philfreo — CHANGES_REQUESTED** [review]
Looking really good – Thanks for all this, really excited about it! Just a few comments.

**Comment by philfreo** [comment]
Rather than changing how `engineering.close.com` works by pointing it to Vercel etc., I'm thinking it may just be easier to set this up from the beginning at `making.close.com`. Then we can change `engineering.close.com` to point to a redirect (S3 bucket probably) like we do for other deprecated domains.

So I think I'm suggesting that you change all references of `engineering.close.com` to `making.close.com` here already before merge.

File: _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md (hunk 5 above)
Comment (craigmdennis) [reply _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md:119]: This is an Eslint thing that's been doing it automatically even though it's set to _add_ them in actual code. I'll try and fix (or at least revert these files; which is what I was doing).

File: _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md (hunk 6 above)
Comment (craigmdennis) [reply _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md:47]: Again, I think this is an eslint/prettier markdown codeblock thing. Will revert.

Comment (buzinas) [reply _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md:47]: This is good, you shouldn't revert :)

File: _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md (hunk 5 above)
Comment (philfreo) [reply _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md:119]: Yeah I figured. Let's figure out why so it stops happening.

File: _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md (hunk 6 above)
Comment (philfreo) [reply _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md:47]: @buzinas You are 100% sure that AWS implementation of XML for S3 Redirection Rules likes this format? If not, then we shouldn't change this.

File: src/pages/404.js (hunk 8 above)
Comment (craigmdennis) [reply src/pages/404.js]: Done. Removed the SVG and inline styles.

File: src/components/navigation.js (hunk 7 above)
Comment (craigmdennis) [reply src/components/navigation.js]: Done.

File: _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md (hunk 6 above)
Comment (craigmdennis) [reply _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md]: Reverted anyway.

File: _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md (hunk 5 above)
Comment (craigmdennis) [reply _posts/2020-02-03-redirects-using-cloudfront-lambda-edge/index.md]: Seems to have stopped now. Reverted some of the files. Some of them have applied JSX formatting.

File: _includes/footer.html (hunk 4 above)
Comment (craigmdennis) [reply _includes/footer.html:27]: Added this using the Segment gatsby plugin.

File: README.md (hunk 3 above)
Comment (craigmdennis) [reply README.md:33]: Added in issue to tackle this separately https://github.com/closeio/engineering.close.com/issues/27

**Pull request merged**